server/src/main/java/org/cloudfoundry/identity/uaa/alias/EntityAliasHandler.java

```diff
@@ -121,53 +120,22 @@ public final T ensureConsistencyOfAliasEntity(
             @NonNull final T originalEntity,
             @Nullable final T existingEntity
     ) throws EntityAliasFailedException {
-        /* If the entity had an alias before the update and the alias feature is now turned off, we break the reference
-         * between the entity and its alias by setting aliasId and aliasZid to null for both of them. Then, all other
-         * changes are only applied to the original entity. */
         final boolean entityHadAlias = existingEntity != null && hasText(existingEntity.getAliasZid());
-        final boolean referenceBreakRequired = entityHadAlias && !aliasEntitiesEnabled;
-        if (referenceBreakRequired) {
-            if (!hasText(existingEntity.getAliasId())) {
-                LOGGER.warn(
-                        "The state of the entity {} before the update had an aliasZid set, but no aliasId.",
-                        existingEntity.getAliasDescription()
-                );
-                return originalEntity;
-            }
-
-            final Optional<T> aliasEntityOpt = retrieveAliasEntity(existingEntity);
-            if (aliasEntityOpt.isEmpty()) {
-                LOGGER.warn(
-                        "The alias referenced in entity {} does not exist, therefore cannot break reference.",
-                        existingEntity.getAliasDescription()
-                );
-                return originalEntity;
-            }
-
-            final T aliasEntity = aliasEntityOpt.get();
-            aliasEntity.setAliasId(null);
-            aliasEntity.setAliasZid(null);
-
-            try {
-                updateEntity(aliasEntity, aliasEntity.getZoneId());
-            } catch (final DataAccessException e) {
-                throw new EntityAliasFailedException(
-                        String.format(
-                                "Could not break reference to alias in entity %s.",
-                                existingEntity.getAliasDescription()
-                        ), HttpStatus.UNPROCESSABLE_ENTITY.value(), e
-                );
-            }
-
-            // no change required in the original entity since its aliasId and aliasZid were already set to null
-            return originalEntity;
+        if (entityHadAlias && !aliasEntitiesEnabled) {
+            // this should already be caught in the validation method
+            throw new IllegalStateException("Performing update on entity with alias while alias feature is disabled.");
```

Comment:
I suggest to document this exception in this method's JavaDoc, i.e. add @throws IllegalStateExceptio with an explanation. Or it appears that we can also throw EntityAliasFailedException instead for this case.

Reply:
fixed with commit ead4176

Reply:
I decided to use `IllegalStateException`s for those cases that should not occur when the `aliasPropertiesAreValid` returned `true`.